### ISSUE TITLE
Instructions on how to run it?

### ISSUE THREAD

**Issue by Petros626** [eric-wieser/ros_numpy#29]
Hey,

i want to convert a sensor_msgs/PointCloud2 to a numpy array. So I play a bag file or a stream which is publishing the message and then run the python script to convert it? Would be more user-friendly if better instructions were given. 

Thank you

**Comment by alvaropascuaI** [comment]
Hi, have you been able to start the repository?

**Comment by Petros626** [comment]
> Hi, have you been able to start the repository?

No 😂

**Comment by alvaropascuaI** [comment]
I have managed to launch data = ros2_numpy.numpify(msg), what I have done has been to first clone the repository and the content of the file is the following and it shows me this in the print of the data
![image](https://user-images.githubusercontent.com/97975075/180164655-54b007be-5eda-4c14-bd80-06007d45f3a9.png)
![image](https://user-images.githubusercontent.com/97975075/180164802-68280c48-8836-4001-9039-3408fa4d8708.png)

**Comment by alvaropascuaI** [comment]
I have been looking at more documentation and some run the following command **sudo apt-get install ros-$release-ros-numpy** try to see and tell me if it worked for you

**Comment by alvaropascuaI** [comment]
What you have to do to get it up and running is to clone the repository and then do the colcon build --packages-select ros2_numpy

**Comment by an99990** [comment]
That is what i did and i got the following 

```
--- stderr: ros2_numpy                         
CMake Error at CMakeLists.txt:14 (find_package):
  By not providing "Findament_cmake_nose.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "ament_cmake_nose", but CMake did not find one.

  Could not find a package configuration file provided by "ament_cmake_nose"
  with any of the following names:

    ament_cmake_noseConfig.cmake
    ament_cmake_nose-config.cmake

  Add the installation prefix of "ament_cmake_nose" to CMAKE_PREFIX_PATH or
  set "ament_cmake_nose_DIR" to a directory containing one of the above
  files.  If "ament_cmake_nose" provides a separate development package or
  SDK, be sure it has been installed.
```

**Comment by alvaropascuaI** [comment]
You have to run this command **sudo apt install ros-foxy-ament-cmake-nose**, when you get more errors like this I recommend that you use **apt search ros-foxy- ...** + the missing library to be able to install it later, in my case the version of ros was foxy you put the one have

**Comment by Genozen** [comment]
> You have to run this command **sudo apt install ros-foxy-ament-cmake-nose**, when you get more errors like this I recommend that you use **apt search ros-foxy- ...** + the missing library to be able to install it later, in my case the version of ros was foxy you put the one have

Getting additional error:
```
CMake Error at CMakeLists.txt:7 (find_package):
  By not providing "Findcatkin.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "catkin", but
  CMake did not find one.

  Could not find a package configuration file provided by "catkin" with any
  of the following names:

    catkinConfig.cmake
    catkin-config.cmake

  Add the installation prefix of "catkin" to CMAKE_PREFIX_PATH or set
  "catkin_DIR" to a directory containing one of the above files.  If "catkin"
  provides a separate development package or SDK, be sure it has been
  installed.
```

Any suggestions? I've Googled this issue and try `qtcreator` but no success.

**Comment by Genozen** [comment]
> 
@alvaropascuaI 
I have cloned the ros2_numpy form this fork:
https://github.com/Box-Robotics/ros2_numpy

After `colcon build`, I wrote a code to `import ros2_numpy.ros2_numpy as rnp` but it says no module named 'ros2_numpy` is found...

**Comment by alvaropascuaI** [comment]
> > You have to run this command **sudo apt install ros-foxy-ament-cmake-nose**, when you get more errors like this I recommend that you use **apt search ros-foxy- ...** + the missing library to be able to install it later, in my case the version of ros was foxy you put the one have
> 
> Getting additional error:
> 
> ```
> CMake Error at CMakeLists.txt:7 (find_package):
>   By not providing "Findcatkin.cmake" in CMAKE_MODULE_PATH this project has
>   asked CMake to find a package configuration file provided by "catkin", but
>   CMake did not find one.
> 
>   Could not find a package configuration file provided by "catkin" with any
>   of the following names:
> 
>     catkinConfig.cmake
>     catkin-config.cmake
> 
>   Add the installation prefix of "catkin" to CMAKE_PREFIX_PATH or set
>   "catkin_DIR" to a directory containing one of the above files.  If "catkin"
>   provides a separate development package or SDK, be sure it has been
>   installed.
> ```
> 
> Any suggestions? I've Googled this issue and try `qtcreator` but no success.

If you have this problem, is because you aren't in the same branch that your ros version

**Comment by Genozen** [comment]
@alvaropascuaI 
Thank you for the quick reply.

I'm trying to run your code via `python3 test.py` with importing `ros2_numpy.ros2_numpy' but no module of  ros2_numpy` is found. 

Did you have to create a launch file instead and run something like `ros2 launch ros2_numpy test.launch.py`?

**Comment by alvaropascuaI** [comment]
No, now i just got my file up and i just needed to run source /opt/ros/foxy/setup.bash and then run the file with python3 name_file.py, give me more details to know how you get your error

**Comment by Genozen** [comment]
> No, now i just got my file up and i just needed to run source /opt/ros/foxy/setup.bash and then run the file with python3 name_file.py, give me more details to know how you get your error

@alvaropascuaI 

That's very strange...

Here are the steps to reproduce my error:

Setup: Ubuntu 22.04, ROS2 Humble

In my `ros2_ws/src` I cloned the repository from https://github.com/Box-Robotics/ros2_numpy
then 
```
cd ~/ros2_ws
colcon build --symlink-install --packages-select ros2_numpy
source /opt/ros/humble/setup.bash
```

then inside the package `/ros2_ws/src/ros2_numpy` I created a script `test.py` with the following code inside:
```
import ros2_numpy.ros2_numpy as rnp

print('test complete')
```
and ran `python3 test.py` in the terminal
which results in the error:
`ModuleNotFoundError: No module named 'ros2_numpy.ros2_numpy`

What's also interesting is when I type `rnp.` my VSCode was able to suggest the methods like `rnp.numpify()`
![image](https://user-images.githubusercontent.com/74474611/211515161-65213e52-0ba1-4bf4-9bb7-ab9fb5660bfd.png)


Below is what my `~/ros2_ws/src/ros2_numpy` directory look like:
![image](https://user-images.githubusercontent.com/74474611/211514695-4bb6aa5d-f538-4ea2-8c44-f55d231c250d.png)

**Comment by Genozen** [comment]
Not sure if this is correct, but I found a work-around by simply calling individual scripts:
![image](https://user-images.githubusercontent.com/74474611/211523969-a72ee902-e405-4e7b-b7cc-d99771aa735b.png)

![image](https://user-images.githubusercontent.com/74474611/211524066-3773da5a-cad0-44cb-9d22-241d4f0ae6de.png)


Hope there's a better way to do it as I'm not sure if I'm missing anything else that's needed (for example the tf_transformation was missing)

**Comment by alvaropascuaI** [comment]
You do that but you need to launch source before the colcon like this

```
cd ~/ros2_ws
source /opt/ros/humble/setup.bash
colcon build --symlink-install --packages-select ros2_numpy
```

**Comment by alvaropascuaI** [comment]
@Genozen Did you resolve with I told you?
